### ISSUE TITLE
GLTFLoader onProgress total is always 0

### ISSUE THREAD

**Issue by santacrab** [mrdoob/three.js#15584]
When I load a model using the GLTFLoader, when I try to calculate the progress % in the onProgress function, the "total" value of the ProgressEvent is always 0. I tried with different models and I always get the same result.

Here my onProgress function:
`function (xhr) {
        console.log((xhr.loaded / xhr.total * 100) + '% loaded');
      }`

xhr.total is always 0 so the result is always "Infinity% loaded". I saw the same thing on many examples online.
Attached you can also find a screen of the ProgressEvent debugged.
<img width="642" alt="schermata 2019-01-15 alle 10 03 48" src="https://user-images.githubusercontent.com/1327320/51169808-17d8be00-18ad-11e9-94d0-8630d8a400ae.png">

##### Three.js version

- [ ] Dev
- [X] r100
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [ ] All of them
- [ ] Windows
- [X] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

**Comment by Mugen87** [comment]
In general, using the `onProgress` callback of `THREE.LoadingManager` makes more sense when loading a `gltf` file, see #13895

**Comment by donmccurdy** [comment]
From the issue, I can't tell if the OP is loading multiple files. If there's only one file, progress events aren't much help...

@santacrab note that onProgress callbacks for all loaders track the number of network requests completed vs the number of network requests made. They don't provide smooth incremental progress while a single file is parsed, for example. Are you loading multiple files?

However, I don't know why `.total` should ever be less than `.loaded` ... could you share a demo?

**Comment by Mugen87** [comment]
>note that onProgress callbacks for all loaders track the number of network requests completed vs the number of network requests made.

I don't think this is true. In the following example `onProgress()` actually tracks how much of the `OBJ` file is already downloaded.


https://threejs.org/examples/webgl_loader_obj

**Comment by donmccurdy** [comment]
Oh good point. GLTFLoader just passes the `onProgress` event into its FileLoader for the initial request, and not for additional resources... I can't think of a better way to handle that at the moment, but open to ideas. In general LoadingManager is probably more helpful, yes.

**Comment by RemusMar** [comment]
Come on Don ...
In the GLTF loader "onProgress" tracks how much of the GLB/GLTF file is already downloaded.
Example: http://necromanthus.com/Test/html5/Lara_gltf_envMap.html

**Comment by Mugen87** [comment]
> I can't think of a better way to handle that at the moment, but open to ideas. In general LoadingManager is probably more helpful, yes.

I think we already talked about this and agreed to refer to `LoadingManager` (sry, can't find the issue^^).

**Comment by santacrab** [comment]
The LoadingManager onProgress works as expected and I can track the total number of files loaded.
Still, I don't get how I always get that 0 value as total, in the onProgress of the GLTFLoader.load().

I'm loading  different gltf files and for all of them, when I inspect the xhr ProgressEvent with the debugger, the total is always 0. I attach one of the models I try to load, if you want to try it out (zipped it to upload).
[Sinius_Helfer.gltf.zip](https://github.com/mrdoob/three.js/files/2763232/Sinius_Helfer.gltf.zip)

**Comment by santacrab** [comment]
Thank you. I'll take a look at the web server configuration!

**Comment by silvainSayduck** [comment]
I made an attempt at implementing a proper onProgress with xhr (as number of files really isn't very helpful in most glTF cases, as file size can vary hugely between meshes and textures...) which might be of interest to some people.

See https://github.com/mrdoob/three.js/pull/16485#issuecomment-514977371

**Comment by cconsta1** [comment]
Hi, 

I'm running three.js using node v12.17.0 (npm v6.14.4). Three.js is part of a react project, and I run the web app using `npm run start`. When I try to load gltf models the `xhr.total` value is 0, and the model doesn't load. If I load obj files using the OBJLoader, everything runs fine. How can I address this problem? 

The same code works just fine if I don't use npm and I instead run the code using VScode's webserver. So I believe that something is wrong with the npm webserver.

**Comment by dweremeichik** [comment]
I just ran into this too; it appears that Snowpack doesn't set the Content-Length. Any chance that this could be added as a note to the documentation for GLTFLoader?

**Comment by donmccurdy** [comment]
@dweremeichik Sorry for the confusion! I think that might be too esoteric for the GLTFLoader docs – three.js has a lot of loaders, and any of those can be deployed with many different development servers (like Snowpack) or production servers.

**Comment by dweremeichik** [comment]
My wording may have been a bit poor. I suppose what I should have said was that it may be good to mention in the docs that this feature requires your webserver to send the content-length header for GLTF files. Otherwise you won't get a total.

**Comment by donmccurdy** [comment]
I see! Yes I think I'd be fine with that change, although I don't know the details of the `onProgress` reporting.

**Comment by mrdoob** [comment]
@dweremeichik Would you like to make a PR with that docs improvement?

**Comment by yunusbayraktaroglu** [comment]
GLTFLoader's ProgressEvent.lengthComputable value returns false, but RGBELoader's true.

Three 0.137.5

**Comment by donmccurdy** [comment]
@yunusbayraktaroglu this property will depend on whether your server sets a Content-Length header, which it may do only for certain asset types. You can check for that header in the Network tab of your developer tools.
